linked-list.rb

```diff
@@ -63,9 +63,34 @@ def display
    end
 
    def include?(key)
+     current = @head
+
+     while current.next_node != nil
+       return true if current.value == key
+       current = current.next_node
+     end
+
+     if current.next_node == nil
+       return true if current.value == key
+     end
+
+     return false
    end
 
    def size
+     current = @head
+     size = 0
+
+     while current.next_node != nil
+       size += 1
+       current = current.next_node
+     end
+
+     if current.next_node == nil
+       size += 1
+     end
+
+     size
    end
```

Comment:
This looks great and works, but in both of these (include?, size) you had to account for that last node separately -- if you change your loop to while current != nil instead of while current.next_node != nil then you wouldn't need that! I told you to model after display which does it the latter way -- because sometimes you need to do something different with the last case and I wanted that to be the pattern I showed -- but now I want to point out how we could have optimized the pattern just slightly. Hopefully that makes sense!